openproductapiclient/client.py

```diff
@@ -0,0 +1,244 @@
+        """
+
+        return client.request(
+            "get",
+            urljoin(base=client.base_url, url=url),
+            params=kwargs,
+        )
+
+    # **************************
+    # Product API endpoints
+    # **************************
+
+    def __products_api_get_request(self, url, **kwargs):
+        """
+        Retrieves products from products API.
+        Filter-arguments can be passed as kwargs.
+        """
+
+        client = self.open_product_api_client
+        response = self.__make_get_request(client, url, **kwargs)
+
+        response.raise_for_status()
+        results = response.json()
+
+        if not results:
+            return "Failed to fetch products API."
+        return results
+
+    def get_products(
+        self,
+        *,
+        aanmaak_datum: str | None = None,
+        aanmaak_datum__gte: str | None = None,
+        aanmaak_datum__lte: str | None = None,
+        dataobject_attr: str | None = None,
+        documenten__uuid: str | None = None,
+        eigenaren__bsn: str | None = None,
+        eigenaren__klantnummer: str | None = None,
+        eigenaren__kvk_nummer: str | None = None,
+        eigenaren__uuid: str | None = None,
+        eigenaren__vestigingsnummer: str | None = None,
+        eind_datum: str | None = None,
+        eind_datum__gte: str | None = None,
+        eind_datum__lte: str | None = None,
+        frequentie: str | None = None,
+        gepubliceerd: bool | None = None,
+        page: int | None = None,
+        page_size: int | None = None,
+        prijs: float | None = None,
+        prijs_gte: float | None = None,
+        prijs_lte: float | None = None,
+        producttype__code: str | None = None,
+        producttype__code__in: list[str] | None = None,
+        producttype__naam: str | None = None,
+        producttype__naam__in: list[str] | None = None,
+        producttype__uuid: str | None = None,
+        producttype__uuid__in: list[str] | None = None,
+        start_datum: str | None = None,
+        start_datum_gte: str | None = None,
+        start_datum_lte: str | None = None,
+        status: str | None = None,
+        uniforme_product_naam: str | None = None,
+        update_datum: str | None = None,
+        update_datum__gte: str | None = None,
+        update_datum__lte: str | None = None,
+        verbruiksobject_attr: str | None = None,
+    ) -> list:
+        """
+        Retrieves products from products API.
+        """
+
+        params = get_filled_params(locals())
+        return self.__products_api_get_request("producten", **params)
+
+    def get_product(self, uuid):
+        """
+        Retrieves a product from products API.
+        Filter-arguments can be passed as kwargs.
+        """
+
+        return self.__products_api_get_request(f"producten/{uuid}/")
+
+    # **************************
+    # Product Type API endpoints
+    # **************************
+
+    def __producttypes_api_get_request(self, url, **kwargs):
```

Comment:
A single underscore will do to indicate it's a private method, double underscores might lead to confusion with Python's dunder methods in a search:

```suggestion
    def _producttypes_api_get_request(self, url, **kwargs):
```